File: Hayaa.BaseModel/Hayaa.BaseModel/BigGridPager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.BaseModel
{
    /// <summary>
    /// 大数据分页数据容器类
    /// </summary>
  public  class BigGridPager<T>: FunctionResult<T> where T:BaseData
    {
        /// <summary>
        /// 当前指向数据页
        /// </summary>
        public long CurrentIndex { set; get; }
        /// <summary>
        /// 数据页尺寸
        /// </summary>
        public long PageSize { set; get; }
        /// <summary>
        /// 总页数
        /// </summary>
        public long PageTotal { set; get; }
    }
}

File: Java/hayaa.basemodel/src/main/java/hayaa/basemodel/model/GridPager/BigGridPager.java
package hayaa.basemodel.model.GridPager;

import hayaa.basemodel.model.BaseData;
import hayaa.basemodel.model.FunctionResult;

public class BigGridPager<T extends BaseData> extends FunctionResult<T> {
    private long CurrentIndex;
    private long PageSize;
    private long PageTotal;

    public long getCurrentIndex() {
        return CurrentIndex;
    }

    public void setCurrentIndex(long currentIndex) {
        CurrentIndex = currentIndex;
    }

    public long getPageSize() {
        return PageSize;
    }

    public void setPageSize(long pageSize) {
        PageSize = pageSize;
    }

    public long getPageTotal() {
        return PageTotal;
    }

    public void setPageTotal(long pageTotal) {
        PageTotal = pageTotal;
    }


}
